--- juno.dts.ori	2015-01-16 14:32:32.992856014 +0100
+++ juno.dts	2015-01-16 16:06:09.889739897 +0100
@@ -106,8 +106,19 @@
 
 	memory@80000000 {
 		device_type = "memory";
+		/*
 		reg = <0x00000000 0x80000000 0x0 0x7f000000>,
 		      <0x00000008 0x80000000 0x1 0x80000000>;
+		*/
+		/*
+		 * Real DRAM0 is 0x80000000 0x80000000
+		 *
+		 * Then we use 0xFEF00000 0x00100000 as shared memory
+		 * and           0xFF000000 0x00FF8000 as tdram
+		 * and		 0xFFFF8000 0x00008000 as reserved for SCP
+		 */
+		reg = <0x00000000 0x80000000 0x0 0x7ef00000>,
+		      <0x00000008 0x80000000 0x1 0x80000000>;
 	};
 
 	/* memory@14000000 {
